src/init.lua

```diff
@@ -1154,9 +1154,10 @@ xplr.config.modes.builtin.default = {
           "PopMode",
           { SwitchModeBuiltin = "rename" },
           {
-            BashExecSilently = [===[
+            BashExecSilently0 = [===[
               NAME=$(basename "${XPLR_FOCUS_PATH:?}")
-              echo SetInputBuffer: "'${NAME:?}'" >> "${XPLR_PIPE_MSG_IN:?}"
+              NAME_ESC=${NAME//\"/\\\"}
+              printf 'SetInputBuffer: "%s"\0' "${NAME_ESC:?}" >> "${XPLR_PIPE_MSG_IN:?}"
```

Comment:
This seems to be broken. I did some tests to see how yaml parses newlines, and it seems like newlines get translated into spaces unless you double escape them:

```python
from yaml import load, Loader

data = 'name: "foo\nbar"'
print(load(data, Loader=Loader)["name"]) # => "foo bar"

data = 'name: "foo\\nbar"'
print(load(data, Loader=Loader)["name"]) # => "foo\nbar"
```

I'm guessing the file name will have to be escaped before setting the input prompt text and unescaped after submitting it, but I thought you might have more insight on what might be happening when going from yaml to rust.

Reply:
I don't think it's yaml encoding, Input buffer can't handle newlines. I don't yet know why it's getting converted into space though, but better than crashing I guess.

I'm not sure if we should allow creating files with newlines. I think such files only get generated by mistake rather than intention. So allowing this might do more harm than good.

Reply:
My bad. We definitely need to escape newlines along with double quotes. But `%q` is not the way, because it will also escape single quotes. Let's go with `NAME_ESC=${NAME_ESC//$'\n'/\\n}`

Reply:
Update: To be handled by new sub command -m